pkg/backend/display/diff.go

```diff
@@ -433,17 +433,22 @@ func renderDiffResourceOutputsEvent(
 ) string {
 	out := &bytes.Buffer{}
 	if shouldShow(payload.Metadata, opts) || isRootStack(payload.Metadata) {
-		// If this is the output step for an import, we actually want to display the diff at this point.
-		if payload.Metadata.Op == deploy.OpImport {
-			renderDiff(out, payload.Metadata, payload.Planning, payload.Debug, seen, opts)
+
+		refresh := false // are these outputs from a refresh?
+		if m, has := seen[payload.Metadata.URN]; has && m.Op == deploy.OpRefresh {
+			refresh = true
+		}
+
+		// If this is the output step for an import or refresh, we actually want to display the diff at this point.
+		if payload.Metadata.Op == deploy.OpImport || (refresh && payload.Metadata.Op == deploy.OpUpdate) {
```

Comment:
For refresh, we now render a normal input diff but at ResourceOutputsEvent time (because we don't know the diff until we run the step).  Very similar to Import.  However, the step gets rewritten into a "OpUpdate" if there are changes, so look at that.

I believe it is right to ignore the OpSame and OpDelete cases that an OpRefresh could also be rewritten to - but it's possible there are cases where those should also go through this path?

Reply:
This probably deserves a little bit more explanation of why we are ignoring `OpSame` and `OpDelete` in a comment.  I think for `OpSame` it makes sense to hide the diff (there shouldn't be one). For `OpDelete` I think it also makes sense because we're deleting, so the diff would be "delete everything", which would would already indicate otherwise?

Reply:
Yep -- will expand upon this.